optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/fact/TestGenPrimitiveValueProvider.java

```diff
@@ -15,6 +15,10 @@
  */
 package org.optaplanner.core.impl.score.director.drools.testgen.fact;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@SuppressFBWarnings(value = "UR_UNINIT_READ", justification = "Uninitialized fields are intentional in this class - "
+        + "to make sure the uninitialized value is the same as JVM would use for uninitialized fields in facts.")
```

Comment:
The default values for primitive values are written in the Java Language Spec 1.0 for over than 20 years.
I think it's safe to hard code them :)

Reply:
@yurloc Would you mind if I change this code according to this:


 ```
       if (value.getClass().equals(Byte.class)) {
            uninitialized = (byte) 0;
        } else if (value.getClass().equals(Short.class)) {
            uninitialized = (short) 0;
        } else if (value.getClass().equals(Integer.class)) {
            uninitialized = 0;
        } else if (value.getClass().equals(Long.class)) {
            uninitialized = 0L;
        } else if (value.getClass().equals(Float.class)) {
            uninitialized = 0.0F;
        } else if (value.getClass().equals(Double.class)) {
            uninitialized = 0.0;
        } else if (value.getClass().equals(Boolean.class)) {
            uninitialized = false;
        } else if (value.getClass().equals(Character.class)) {
            uninitialized = (char) 0;
```

Reply:
OK